include/control_filters/low_pass_filter.hpp

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2023, Stogl Robotics Consulting UG (haftungsbeschränkt)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CONTROL_FILTERS__LOW_PASS_FILTER_HPP_
+#define CONTROL_FILTERS__LOW_PASS_FILTER_HPP_
+
+#include <Eigen/Dense>
+#include <cmath>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "low_pass_filter_parameters.hpp"
+#include "filters/filter_base.hpp"
+
+#include "geometry_msgs/msg/wrench_stamped.hpp"
+
+namespace control_filters
+{
+
+template <typename T>
+class LowPassFilter : public filters::FilterBase<T>
+{
+public:
+  LowPassFilter();
+
+  ~LowPassFilter() override;
+
+  bool configure() override;
+
+  bool update(const T & data_in, T & data_out) override;
+
+protected:
+  void compute_internal_params()
+  {
+    a1_ = exp(
+      -1.0 / parameters_.sampling_frequency * (2.0 * M_PI * parameters_.damping_frequency) /
+      (pow(10.0, parameters_.damping_intensity / -10.0)));
+    b1_ = 1.0 - a1_;
+  };
+
+private:
+  rclcpp::Clock::SharedPtr clock_;
+  std::shared_ptr<rclcpp::Logger> logger_;
+  std::shared_ptr<low_pass_filter::ParamListener> parameter_handler_;
+  low_pass_filter::Params parameters_;
+
+  // Filter parameters
+  // TODO(destogl): we should do this more intelligently using only one set of types
```

Comment:
Not necessary since we didn't come up with any better solution.

```suggestion
```

Reply:
This was not worked on yet. I had a quick look, maybe something along using std::enable_if to detect if a type can be directly filtered with the y(n)= b * x(n-1) + a * y(n-1) example double, ints Eigen<N,1>, etc..) or if a specialization has been provided for converting the T type to a processable storage...
https://medium.com/@sidbhasin82/c-templates-what-is-std-enable-if-and-how-to-use-it-fd76d3abbabe